schema_based_on_diagram.sql

```diff
@@ -0,0 +1,37 @@
+CREATE SCHEMA clinic;
+
+CREATE TABLE patients(
+    id INT SERIAL PRIMARY KEY,
+    name VARCHAR(100),
+    date_of_birth DATE
+);
+
+CREATE TABLE treatments(
+    id INT SERIAL PRIMARY KEY,
+    type VARCHAR(100),
+    name VARCHAR(100)
+);
+
+CREATE TABLE medical_histories(
+    id INT SERIAL PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patient_id INT references patients(id),
+    status VARCHAR(100)
+);
+
+CREATE TABLE invoices(
+    id INT SERIAL PRIMARY KEY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT references medical_histories(id)
+);
+
+CREATE TABLE invoice_items(
+    id INT SERIAL PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity int,
+    total_price DECIMAL,
+    invoice_id INT FOREIGN KEY references invoices(id),
+    treatment_id INT FOREIGN KEY references treatments(id)
+);
```

Comment:
- [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md)for this project,  "`Join tables from many-to-many relationships might not appear in the diagram, but you still need them.`", a "join table" for the many-to-many relastionships should be created, even if they are not showed on the diagram. Therefore you should create a table between medical_histories and treatments, something like so:   

```sql
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

- [ ] As it s a [requirement ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md)for this project, "`Remember to add the FK indexes.`", please make sure all FOREIGN KEYS have indexes created, something like so: 

* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`

##### Please kindly fix that.

Reply:
owh. that's amazing. thank you for your great review! let me check it out and apply this to improve my code.
Thanks alot.